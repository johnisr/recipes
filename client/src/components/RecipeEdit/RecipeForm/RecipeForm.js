import React, { Component } from 'react';
import { reduxForm, Field, FieldArray } from 'redux-form';
import { Link } from 'react-router-dom';

import RecipeField from './RecipeField/RecipeField';
import RecipeFieldTime from './RecipeFieldTime/RecipeFieldTime';
import RecipeFieldArray from './RecipeFieldArray/RecipeFieldArray';

export class RecipeForm extends Component {
  state = { showAllErrors: false };
  onSubmit = e => {
    e.preventDefault();
    const { onRecipeSubmit, valid } = this.props;
    if (valid) {
      onRecipeSubmit();
    } else {
      this.setState(() => ({ showAllErrors: true }));
    }
  };
  render() {
    const { submitting } = this.props;
    return (
      <div className="row">
        <form onSubmit={e => this.onSubmit(e)}>
          <Field
            name="name"
            component={RecipeField}
            type="text"
            label="Name"
            showAllErrors={this.state.showAllErrors}
          />
          <Field
            name="summary"
            component={RecipeField}
            type="text"
            label="Summary"
            showAllErrors={this.state.showAllErrors}
          />
          <Field
            name="notes"
            component={RecipeField}
            type="textarea"
            label="Notes Section"
            showAllErrors={this.state.showAllErrors}
          />
          <div className="row">
            <div className="col s4">
              <FieldArray
                classname="col s4"
                name="cookingTime"
                component={RecipeFieldTime}
                type="time"
                label="Cooking Time"
                showAllErrors={this.state.showAllErrors}
              />
            </div>
            <div className="col s4">
              <FieldArray
                classname="col s4"
                name="preparationTime"
                component={RecipeFieldTime}
                type="time"
                label="Preparation Time"
                showAllErrors={this.state.showAllErrors}
              />
            </div>
            <div className="col s4">
              <Field
                classname="col s4"
                name="category"
                component={RecipeField}
                type="textarea"
                label="Categories"
                showAllErrors={this.state.showAllErrors}
              />
            </div>
          </div>
          <div style={{ marginBottom: '60px' }}>
            <FieldArray
              name="ingredients"
              required
              component={RecipeFieldArray}
              label="Ingredients"
              sublabel="Ingredients"
              showAllErrors={this.state.showAllErrors}
            />
          </div>
          <div style={{ marginBottom: '60px' }}>
            <FieldArray
              name="preparation"
              component={RecipeFieldArray}
              label="Preparation"
              sublabel="Instructions"
              showAllErrors={this.state.showAllErrors}
            />
          </div>
          <div>
            <FieldArray
              name="cooking"
              required
              component={RecipeFieldArray}
              label="Cooking"
              sublabel="Instructions"
              showAllErrors={this.state.showAllErrors}
            />
          </div>
          <Link to="/dashboard" className="red btn-flat white-text">
            Cancel
          </Link>
          <button
            className="teal btn-flat right white-text"
            disabled={submitting}
            type="submit"
          >
            Next
            <i className="material-icons right">done</i>
          </button>
        </form>
      </div>
    );
  }
}

const validate = values => {
  const errors = {};

  if (!values.name) {
    errors.name = 'Required';
  }

  if (!values.ingredients || !values.ingredients.length) {
    errors.ingredients = {
      _error: 'At least one ingredient section must be entered',
    };
  } else {
    const ingredientArrayErrors = [];
    values.ingredients.forEach((ingredient, index) => {
      const ingredientErrors = {};
      if (!ingredient || !ingredient.body) {
        ingredientErrors.body = 'Required';
        ingredientArrayErrors[index] = ingredientErrors;
      }

      return ingredientErrors;
    });
    if (ingredientArrayErrors.length) {
      errors.ingredients = ingredientArrayErrors;
    }
  }
  if (!values.cooking || !values.cooking.length) {
    errors.cooking = {
      _error: 'At least one cooking section must be entered',
    };
  } else {
    const cookingArrayErrors = [];
    values.cooking.forEach((ingredient, index) => {
      const cookingErrors = {};
      if (!ingredient || !ingredient.body) {
        cookingErrors.body = 'Required';
        cookingArrayErrors[index] = cookingErrors;
      }

      return cookingErrors;
    });
    if (cookingArrayErrors.length) {
      errors.cooking = cookingArrayErrors;
    }
  }

  return errors;
};

export default reduxForm({
  validate,
  form: 'recipeForm',
  destroyOnUnmount: false,
})(RecipeForm);
