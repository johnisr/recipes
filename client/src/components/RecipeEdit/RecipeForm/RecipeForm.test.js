import React from 'react';
import { shallow } from 'enzyme';
import { RecipeForm } from './RecipeForm';

let valid;
let onRecipeSubmitSpy;
let submitting;
beforeAll(() => {
  valid = false;
  onRecipeSubmitSpy = jest.fn();
  submitting = false;
});

test('should render correctly', () => {
  const wrapper = shallow(
    <RecipeForm
      valid={valid}
      onRecipeSubmit={onRecipeSubmitSpy}
      submitting={submitting}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should disable button when submitting is true', () => {
  submitting = true;
  const wrapper = shallow(
    <RecipeForm
      valid={valid}
      onRecipeSubmit={onRecipeSubmitSpy}
      submitting={submitting}
    />
  );

  expect(
    wrapper
      .find('Button')
      .at(1)
      .prop('disabled')
  ).toBe(true);
});

test('should call onRecipeSubmit if form valid (true)', () => {
  valid = true;
  submitting = false;
  const wrapper = shallow(
    <RecipeForm
      valid={valid}
      onRecipeSubmit={onRecipeSubmitSpy}
      submitting={submitting}
    />
  );
  // Semantic UI takes Form => Object with prop "as": "form"
  wrapper
    .find({ as: 'form' })
    .at(0)
    .simulate('submit', {
      preventDefault: () => {},
    });

  expect(onRecipeSubmitSpy).toHaveBeenCalled();
});

test('should set showAllErrors if form not valid', () => {
  valid = false;
  submitting = false;
  const wrapper = shallow(
    <RecipeForm
      valid={valid}
      onRecipeSubmit={onRecipeSubmitSpy}
      submitting={submitting}
    />
  );

  // Semantic UI takes Form => Object with prop "as": "form"
  wrapper
    .find({ as: 'form' })
    .at(0)
    .simulate('submit', {
      preventDefault: () => {},
    });
  expect(wrapper.state('showAllErrors')).toBe(true);
});
