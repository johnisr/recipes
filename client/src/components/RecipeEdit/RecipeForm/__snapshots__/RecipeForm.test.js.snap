// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeForm
    onRecipeSubmit={[Function]}
    submitting={false}
    valid={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <Field
                component={[Function]}
                label="Name"
                name="name"
                showAllErrors={false}
                type="text"
        />
        <Field
                component={[Function]}
                label="Summary"
                name="summary"
                showAllErrors={false}
                type="text"
        />
        <Field
                component={[Function]}
                label="Notes Section"
                name="notes"
                showAllErrors={false}
                type="textarea"
        />
        <div
                className="row"
        >
                <div
                        className="col s4"
                >
                        <FieldArray
                                classname="col s4"
                                component={[Function]}
                                label="Cooking Time"
                                name="cookingTime"
                                showAllErrors={false}
                                type="time"
                        />
                </div>
                <div
                        className="col s4"
                >
                        <FieldArray
                                classname="col s4"
                                component={[Function]}
                                label="Preparation Time"
                                name="preparationTime"
                                showAllErrors={false}
                                type="time"
                        />
                </div>
                <div
                        className="col s4"
                >
                        <Field
                                classname="col s4"
                                component={[Function]}
                                label="Categories"
                                name="category"
                                showAllErrors={false}
                                type="textarea"
                        />
                </div>
        </div>
        <div
                style={
                        Object {
                                "marginBottom": "60px",
                              }
                }
        >
                <FieldArray
                        component={[Function]}
                        label="Ingredients"
                        name="ingredients"
                        required={true}
                        showAllErrors={false}
                        sublabel="Ingredients"
                />
        </div>
        <div
                style={
                        Object {
                                "marginBottom": "60px",
                              }
                }
        >
                <FieldArray
                        component={[Function]}
                        label="Preparation"
                        name="preparation"
                        showAllErrors={false}
                        sublabel="Instructions"
                />
        </div>
        <div>
                <FieldArray
                        component={[Function]}
                        label="Cooking"
                        name="cooking"
                        required={true}
                        showAllErrors={false}
                        sublabel="Instructions"
                />
        </div>
        <Link
                className="red btn-flat white-text"
                replace={false}
                to="/dashboard"
        >
                Cancel
        </Link>
        <button
                className="teal btn-flat right white-text"
                disabled={false}
                type="submit"
        >
                Next
                <i
                        className="material-icons right"
                >
                        done
                </i>
        </button>
</form>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Field
            component={[Function]}
            label="Name"
            name="name"
            showAllErrors={false}
            type="text"
/>,
          <Field
            component={[Function]}
            label="Summary"
            name="summary"
            showAllErrors={false}
            type="text"
/>,
          <Field
            component={[Function]}
            label="Notes Section"
            name="notes"
            showAllErrors={false}
            type="textarea"
/>,
          <div
            className="row"
>
            <div
                        className="col s4"
            >
                        <FieldArray
                                    classname="col s4"
                                    component={[Function]}
                                    label="Cooking Time"
                                    name="cookingTime"
                                    showAllErrors={false}
                                    type="time"
                        />
            </div>
            <div
                        className="col s4"
            >
                        <FieldArray
                                    classname="col s4"
                                    component={[Function]}
                                    label="Preparation Time"
                                    name="preparationTime"
                                    showAllErrors={false}
                                    type="time"
                        />
            </div>
            <div
                        className="col s4"
            >
                        <Field
                                    classname="col s4"
                                    component={[Function]}
                                    label="Categories"
                                    name="category"
                                    showAllErrors={false}
                                    type="textarea"
                        />
            </div>
</div>,
          <div
            style={
                        Object {
                                    "marginBottom": "60px",
                                  }
            }
>
            <FieldArray
                        component={[Function]}
                        label="Ingredients"
                        name="ingredients"
                        required={true}
                        showAllErrors={false}
                        sublabel="Ingredients"
            />
</div>,
          <div
            style={
                        Object {
                                    "marginBottom": "60px",
                                  }
            }
>
            <FieldArray
                        component={[Function]}
                        label="Preparation"
                        name="preparation"
                        showAllErrors={false}
                        sublabel="Instructions"
            />
</div>,
          <div>
            <FieldArray
                        component={[Function]}
                        label="Cooking"
                        name="cooking"
                        required={true}
                        showAllErrors={false}
                        sublabel="Instructions"
            />
</div>,
          <Link
            className="red btn-flat white-text"
            replace={false}
            to="/dashboard"
>
            Cancel
</Link>,
          <button
            className="teal btn-flat right white-text"
            disabled={false}
            type="submit"
>
            Next
            <i
                        className="material-icons right"
            >
                        done
            </i>
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "label": "Name",
            "name": "name",
            "showAllErrors": false,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "label": "Summary",
            "name": "summary",
            "showAllErrors": false,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "label": "Notes Section",
            "name": "notes",
            "showAllErrors": false,
            "type": "textarea",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col s4"
>
                <FieldArray
                                classname="col s4"
                                component={[Function]}
                                label="Cooking Time"
                                name="cookingTime"
                                showAllErrors={false}
                                type="time"
                />
</div>,
              <div
                className="col s4"
>
                <FieldArray
                                classname="col s4"
                                component={[Function]}
                                label="Preparation Time"
                                name="preparationTime"
                                showAllErrors={false}
                                type="time"
                />
</div>,
              <div
                className="col s4"
>
                <Field
                                classname="col s4"
                                component={[Function]}
                                label="Categories"
                                name="category"
                                showAllErrors={false}
                                type="textarea"
                />
</div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FieldArray
                  classname="col s4"
                  component={[Function]}
                  label="Cooking Time"
                  name="cookingTime"
                  showAllErrors={false}
                  type="time"
/>,
                "className": "col s4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "classname": "col s4",
                  "component": [Function],
                  "label": "Cooking Time",
                  "name": "cookingTime",
                  "showAllErrors": false,
                  "type": "time",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FieldArray
                  classname="col s4"
                  component={[Function]}
                  label="Preparation Time"
                  name="preparationTime"
                  showAllErrors={false}
                  type="time"
/>,
                "className": "col s4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "classname": "col s4",
                  "component": [Function],
                  "label": "Preparation Time",
                  "name": "preparationTime",
                  "showAllErrors": false,
                  "type": "time",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Field
                  classname="col s4"
                  component={[Function]}
                  label="Categories"
                  name="category"
                  showAllErrors={false}
                  type="textarea"
/>,
                "className": "col s4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "classname": "col s4",
                  "component": [Function],
                  "label": "Categories",
                  "name": "category",
                  "showAllErrors": false,
                  "type": "textarea",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FieldArray
              component={[Function]}
              label="Ingredients"
              name="ingredients"
              required={true}
              showAllErrors={false}
              sublabel="Ingredients"
/>,
            "style": Object {
              "marginBottom": "60px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Ingredients",
              "name": "ingredients",
              "required": true,
              "showAllErrors": false,
              "sublabel": "Ingredients",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FieldArray
              component={[Function]}
              label="Preparation"
              name="preparation"
              showAllErrors={false}
              sublabel="Instructions"
/>,
            "style": Object {
              "marginBottom": "60px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Preparation",
              "name": "preparation",
              "showAllErrors": false,
              "sublabel": "Instructions",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FieldArray
              component={[Function]}
              label="Cooking"
              name="cooking"
              required={true}
              showAllErrors={false}
              sublabel="Instructions"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Cooking",
              "name": "cooking",
              "required": true,
              "showAllErrors": false,
              "sublabel": "Instructions",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Cancel",
            "className": "red btn-flat white-text",
            "replace": false,
            "to": "/dashboard",
          },
          "ref": null,
          "rendered": "Cancel",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Next",
              <i
                className="material-icons right"
>
                done
</i>,
            ],
            "className": "teal btn-flat right white-text",
            "disabled": false,
            "type": "submit",
          },
          "ref": null,
          "rendered": Array [
            "Next",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "done",
                "className": "material-icons right",
              },
              "ref": null,
              "rendered": "done",
              "type": "i",
            },
          ],
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <Field
                    component={[Function]}
                    label="Name"
                    name="name"
                    showAllErrors={false}
                    type="text"
          />
          <Field
                    component={[Function]}
                    label="Summary"
                    name="summary"
                    showAllErrors={false}
                    type="text"
          />
          <Field
                    component={[Function]}
                    label="Notes Section"
                    name="notes"
                    showAllErrors={false}
                    type="textarea"
          />
          <div
                    className="row"
          >
                    <div
                              className="col s4"
                    >
                              <FieldArray
                                        classname="col s4"
                                        component={[Function]}
                                        label="Cooking Time"
                                        name="cookingTime"
                                        showAllErrors={false}
                                        type="time"
                              />
                    </div>
                    <div
                              className="col s4"
                    >
                              <FieldArray
                                        classname="col s4"
                                        component={[Function]}
                                        label="Preparation Time"
                                        name="preparationTime"
                                        showAllErrors={false}
                                        type="time"
                              />
                    </div>
                    <div
                              className="col s4"
                    >
                              <Field
                                        classname="col s4"
                                        component={[Function]}
                                        label="Categories"
                                        name="category"
                                        showAllErrors={false}
                                        type="textarea"
                              />
                    </div>
          </div>
          <div
                    style={
                              Object {
                                        "marginBottom": "60px",
                                      }
                    }
          >
                    <FieldArray
                              component={[Function]}
                              label="Ingredients"
                              name="ingredients"
                              required={true}
                              showAllErrors={false}
                              sublabel="Ingredients"
                    />
          </div>
          <div
                    style={
                              Object {
                                        "marginBottom": "60px",
                                      }
                    }
          >
                    <FieldArray
                              component={[Function]}
                              label="Preparation"
                              name="preparation"
                              showAllErrors={false}
                              sublabel="Instructions"
                    />
          </div>
          <div>
                    <FieldArray
                              component={[Function]}
                              label="Cooking"
                              name="cooking"
                              required={true}
                              showAllErrors={false}
                              sublabel="Instructions"
                    />
          </div>
          <Link
                    className="red btn-flat white-text"
                    replace={false}
                    to="/dashboard"
          >
                    Cancel
          </Link>
          <button
                    className="teal btn-flat right white-text"
                    disabled={false}
                    type="submit"
          >
                    Next
                    <i
                              className="material-icons right"
                    >
                              done
                    </i>
          </button>
</form>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Field
              component={[Function]}
              label="Name"
              name="name"
              showAllErrors={false}
              type="text"
/>,
            <Field
              component={[Function]}
              label="Summary"
              name="summary"
              showAllErrors={false}
              type="text"
/>,
            <Field
              component={[Function]}
              label="Notes Section"
              name="notes"
              showAllErrors={false}
              type="textarea"
/>,
            <div
              className="row"
>
              <div
                            className="col s4"
              >
                            <FieldArray
                                          classname="col s4"
                                          component={[Function]}
                                          label="Cooking Time"
                                          name="cookingTime"
                                          showAllErrors={false}
                                          type="time"
                            />
              </div>
              <div
                            className="col s4"
              >
                            <FieldArray
                                          classname="col s4"
                                          component={[Function]}
                                          label="Preparation Time"
                                          name="preparationTime"
                                          showAllErrors={false}
                                          type="time"
                            />
              </div>
              <div
                            className="col s4"
              >
                            <Field
                                          classname="col s4"
                                          component={[Function]}
                                          label="Categories"
                                          name="category"
                                          showAllErrors={false}
                                          type="textarea"
                            />
              </div>
</div>,
            <div
              style={
                            Object {
                                          "marginBottom": "60px",
                                        }
              }
>
              <FieldArray
                            component={[Function]}
                            label="Ingredients"
                            name="ingredients"
                            required={true}
                            showAllErrors={false}
                            sublabel="Ingredients"
              />
</div>,
            <div
              style={
                            Object {
                                          "marginBottom": "60px",
                                        }
              }
>
              <FieldArray
                            component={[Function]}
                            label="Preparation"
                            name="preparation"
                            showAllErrors={false}
                            sublabel="Instructions"
              />
</div>,
            <div>
              <FieldArray
                            component={[Function]}
                            label="Cooking"
                            name="cooking"
                            required={true}
                            showAllErrors={false}
                            sublabel="Instructions"
              />
</div>,
            <Link
              className="red btn-flat white-text"
              replace={false}
              to="/dashboard"
>
              Cancel
</Link>,
            <button
              className="teal btn-flat right white-text"
              disabled={false}
              type="submit"
>
              Next
              <i
                            className="material-icons right"
              >
                            done
              </i>
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Name",
              "name": "name",
              "showAllErrors": false,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Summary",
              "name": "summary",
              "showAllErrors": false,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Notes Section",
              "name": "notes",
              "showAllErrors": false,
              "type": "textarea",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col s4"
>
                  <FieldArray
                                    classname="col s4"
                                    component={[Function]}
                                    label="Cooking Time"
                                    name="cookingTime"
                                    showAllErrors={false}
                                    type="time"
                  />
</div>,
                <div
                  className="col s4"
>
                  <FieldArray
                                    classname="col s4"
                                    component={[Function]}
                                    label="Preparation Time"
                                    name="preparationTime"
                                    showAllErrors={false}
                                    type="time"
                  />
</div>,
                <div
                  className="col s4"
>
                  <Field
                                    classname="col s4"
                                    component={[Function]}
                                    label="Categories"
                                    name="category"
                                    showAllErrors={false}
                                    type="textarea"
                  />
</div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FieldArray
                    classname="col s4"
                    component={[Function]}
                    label="Cooking Time"
                    name="cookingTime"
                    showAllErrors={false}
                    type="time"
/>,
                  "className": "col s4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "classname": "col s4",
                    "component": [Function],
                    "label": "Cooking Time",
                    "name": "cookingTime",
                    "showAllErrors": false,
                    "type": "time",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FieldArray
                    classname="col s4"
                    component={[Function]}
                    label="Preparation Time"
                    name="preparationTime"
                    showAllErrors={false}
                    type="time"
/>,
                  "className": "col s4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "classname": "col s4",
                    "component": [Function],
                    "label": "Preparation Time",
                    "name": "preparationTime",
                    "showAllErrors": false,
                    "type": "time",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Field
                    classname="col s4"
                    component={[Function]}
                    label="Categories"
                    name="category"
                    showAllErrors={false}
                    type="textarea"
/>,
                  "className": "col s4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "classname": "col s4",
                    "component": [Function],
                    "label": "Categories",
                    "name": "category",
                    "showAllErrors": false,
                    "type": "textarea",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FieldArray
                component={[Function]}
                label="Ingredients"
                name="ingredients"
                required={true}
                showAllErrors={false}
                sublabel="Ingredients"
/>,
              "style": Object {
                "marginBottom": "60px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "label": "Ingredients",
                "name": "ingredients",
                "required": true,
                "showAllErrors": false,
                "sublabel": "Ingredients",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FieldArray
                component={[Function]}
                label="Preparation"
                name="preparation"
                showAllErrors={false}
                sublabel="Instructions"
/>,
              "style": Object {
                "marginBottom": "60px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "label": "Preparation",
                "name": "preparation",
                "showAllErrors": false,
                "sublabel": "Instructions",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FieldArray
                component={[Function]}
                label="Cooking"
                name="cooking"
                required={true}
                showAllErrors={false}
                sublabel="Instructions"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "label": "Cooking",
                "name": "cooking",
                "required": true,
                "showAllErrors": false,
                "sublabel": "Instructions",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancel",
              "className": "red btn-flat white-text",
              "replace": false,
              "to": "/dashboard",
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Next",
                <i
                  className="material-icons right"
>
                  done
</i>,
              ],
              "className": "teal btn-flat right white-text",
              "disabled": false,
              "type": "submit",
            },
            "ref": null,
            "rendered": Array [
              "Next",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "done",
                  "className": "material-icons right",
                },
                "ref": null,
                "rendered": "done",
                "type": "i",
              },
            ],
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
